src/storage/mvcc/reader/scanner/delta.rs

```diff
@@ -0,0 +1,247 @@
+    fn next_entry(&mut self) -> TxnResult<Option<TxnEntry>> {
+        Ok(self.read_next()?)
+    }
+    fn take_statistics(&mut self) -> Statistics {
+        std::mem::replace(&mut self.statistics, Statistics::default())
+    }
+}
+
+impl<S: Snapshot> DeltaScanner<S> {
+    pub fn new(
+        cfg: ScannerConfig<S>,
+        lock_cursor: Option<Cursor<S::Iter>>,
+        write_cursor: Cursor<S::Iter>,
+        begin_ts: u64,
+    ) -> DeltaScanner<S> {
+        DeltaScanner {
+            cfg,
+            lock_cursor,
+            write_cursor,
+            statistics: Statistics::default(),
+            default_cursor: None,
+            is_started: false,
+            begin_ts,
+        }
+    }
+    pub fn read_next(&mut self) -> Result<Option<TxnEntry>> {
+        self.check_and_start()?;
+        if self.lock_cursor.is_none() {
+            return self.read_next_write_key();
+        }
+        // TODO: read next with RR isolation
+        unimplemented!()
+    }
+
+    fn read_next_write_key(&mut self) -> Result<Option<TxnEntry>> {
+        loop {
+            if !self.write_cursor.valid()? {
+                return Ok(None);
+            }
+            let w_key = self.write_cursor.key(&mut self.statistics.write);
+            let (user_key, commit_ts) = Key::split_on_ts_for(w_key)?;
+            // Use `from_encoded_slice` to reserve space for ts, so later we can append ts to
+            // the key or its clones without reallocation.
+            let current_user_key = Key::from_encoded_slice(user_key);
+            // goto next key
+            if commit_ts <= self.begin_ts {
+                self.move_write_cursor_to_next_user_key(&current_user_key)?;
+                continue;
+            }
+            if commit_ts <= self.cfg.ts {
+                let v = self.load_data_and_write(&current_user_key)?;
+                self.write_cursor.next(&mut self.statistics.write);
+                return Ok(Some(v));
+            }
+            self.write_cursor.next(&mut self.statistics.write);
+        }
+    }
+
+    /// Load the value by the given `write`. If value is carried in `write`, it will be returned
+    /// directly. Otherwise there will be a default CF look up.
+    ///
+    /// The implementation is the same as `PointGetter::load_data_and_write`.
+    #[inline]
+    fn load_data_and_write(&mut self, user_key: &Key) -> Result<TxnEntry> {
+        let write = Write::parse(self.write_cursor.value(&mut self.statistics.write))?;
+        self.statistics.write.processed += 1;
+        let w = (
+            self.write_cursor.key(&mut self.statistics.write).to_vec(),
+            self.write_cursor.value(&mut self.statistics.write).to_vec(),
+        );
+        // Not PUT or Value is carried in `write`.
+        if write.write_type != WriteType::Put || write.short_value.is_some() {
+            return Ok(TxnEntry::Commit {
+                default: (Vec::new(), Vec::new()),
+                write: w,
+            });
+        };
```

Comment:
After https://github.com/tikv/tikv/pull/5575 , short value might occur in Rollback records to indicate the rollback is protected. Does that matter?
@sticnarf By the way, it just come to my mind that maybe we have code that assumes short value exists only in Put. Maybe we need to take some time to check it.

Reply:
OK.

Reply:
It doesn't matter. In my mind, CDC also needs rollback records.

I've searched the codebase for `short_value`. I think there are no problems. They all ensure it's a PUT.